spec/api.yaml

```diff
@@ -261,28 +267,33 @@ paths:
             application/json:
               schema:
                 type: string
-  /difficulty:
+  /challenge:
```

Comment:
maybe not applicable at this time, but are challenges unique for each request? if I make two distinct calls to `GET /challenge` will I receive two distinct `hash` values? 

on a similar note, should calls to `GET /challenge` include the client's DID as to be associated to the given DID? for example, maybe in the form of a DPoP header?

Reply:
> maybe not applicable at this time, but are challenges unique for each request? if I make two distinct calls to GET /challenge will I receive two distinct hash values?

It is up to the server. If, for example, we use the current block hash as an input, the answer would be no for a period of about 10m. I do not think this is a problem, as the biggest consequence would be receiving another solution within the 10m window which would at most extend the retention period by 10m.

Let me know if you think otherwise.

> on a similar note, should calls to GET /challenge include the client's DID as to be associated to the given DID? for example, maybe in the form of a DPoP header?

Is this necessary? I do need to add a section on verifying the solution, which includes using the requester's DID.

Reply:
there's a potential attack wherein, if the hash is the same across a 10m period, and the hash's aren't associated to a specific DID that, they could find the solution and then spam the system for the next 10-N minutes creating/updating a bunch of DIDs but only solving the challenge once

Reply:
oh - the hashes are associated to a particular DID -- see the algorithm for generating a retention solution https://github.com/TBD54566975/did-dht-method/pull/219/files#diff-c4795c497b83a8c03e33535caf0fb0e1512cecd8cb448f62467326277c152afeR740